src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -270,7 +453,9 @@ public void OnVolumeChanged(float volume)
 		{
 			Player.SeekTo((long)position.TotalMilliseconds);
 
-			await seekToTaskCompletionSource.Task.WaitAsync(token);
+			// Here, we don't want to throw an exception
+			// and to keep the execution on the thread that called this method
+			await seekToTaskCompletionSource.Task.WaitAsync(TimeSpan.FromMinutes(2), token).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing | ConfigureAwaitOptions.ContinueOnCapturedContext);
```

Comment:
This TCS is completed just in one case, so in order to avoid an eternal deadlock I added the timeout there. For more context on this subject just look [here](https://devblogs.microsoft.com/pfxteam/dont-forget-to-complete-your-tasks/)